WG_DYNAMIC_DATA_AND_CAPABILITIES.md

```diff
@@ -0,0 +1,224 @@
+    *   Beginner
+    *   Intermediate
+    *   Advanced
+*   Video perhaps, but only for events
+
+#### 1.9. (DX) The working group has partnered with a third party on development of a production-quality collaborative application (P4)
+
+#### 1.10. (DX) peer-base interfaces and types are formally defined (language agnostic) (P4)
+
+Notes:
+*   Discuss this with libp2p and IPFS teams.
+*   Check out WebIDL.
+*   Try to make this more iterative (not waterfall) as peer-base stabilizes so that we can start implementing some of these components in other languages / platforms.
+
+#### 1.11. (DX) Identity Manager PoC v0.1 is released and is easy to integrate into an existing web DApp (well tested and documented) (P4)
+
+
+#### 1.12. (DX/UX) Chart a collection of UX patterns for DApps (P4)
+
+Notes:
+
+*   Also depends on developers and users to gain literacy about the decentralised concepts
+
+#### 1.13. (DX) Research and document tooling to help with e2e testing for DApps (P3)
+
+### 2. UX
+
+#### 2.1. (UX) An evolving spec for human-friendly way to identify users in DApps was debated and approved - IDM Spec v0.1 (P1)
+
+IDM = Identity Manager
+
+#### 2.2. (UX) A PoC reference implementation for the IDM Spec exists (P2)
+
+Note: (Consider also having a milestone for releasing an IDM MVP).
+
+#### 2.3. (UX) Can replicate app/domain storage using Identity
+
+(Depends on M2.1)
+
+*   It's expected that the same user uses the same DApps on different devices. Research and implement a way for application data produced by the same identity (user) to be synced among devices.
+*   It should support multi-writer scenarios
+
+#### 2.4. (UX) It is possible to build web DApps based on peer-base with good performance attributes (P0)
+
+#### Performance
+
+*   Reduce the number of IPFS nodes when running multiple DApps
+*   Prevent main thread blocking when:
+    *   Doing crypto stuff, perhaps offload to web workers
+    *   Accessing IndexedDB
+*   Explore: using WebRTC as an alternative web sockets?
+*   Replace WS-Star with:
+    *   Rendez-vous
+    *   Circuit-relay
+*   Suggestion: Support Companion and libdweb (libp2p p2p streams over WS + feed of discovery events, exposed by IPFS daemon)
+*   **Required: Measure** (automated) and iterate
+
+#### Reliability
+
+*   Solve concurrency problems when two tabs use the same IPFS node
+
+### 3. DApps
+
+### 3.1
+
+#### 3.1.1 (DApps) - PeerPad used as the primary note taking tool inside PL (not be at feature parity with Cryptpad but "good enough", internal users do not rebel) (P1)
+
+#### 3.1.2 (DApps) - PeerPad has comprehensive E2E/load tests (P1)
+
+*   PP tests use common PL testbed (being developed currently by libp2p team) // _cross-team dependency_
+*   range of scenarios tested  (single player, normal WG scale, single bad actor, overwhelming load that shows degenerate behavior, etc)
+
+#### 3.1.3 (DApps/UX) - Comprehensive competitive *-Pad field study completed  (P2)
```

Comment:
It shouldn't be "competition study", rather, what is out there and which features do those products provide that we currently don't have a solution in our DWeb Stack to make it possible

Reply:
@parkan could you review this?